File: quarkus-tryout-java/src/main/java/org/quarkus/book/ExampleResource.java
package org.quarkus.book;

import jakarta.ws.rs.GET;
import jakarta.ws.rs.Path;
import jakarta.ws.rs.Produces;
import jakarta.ws.rs.core.MediaType;
import org.eclipse.microprofile.config.inject.ConfigProperty;

@Path("/hello")
public class ExampleResource {
  @ConfigProperty(name = "books.genre", defaultValue = "Science Fiction")
  String genre;


  @GET
  @Produces(MediaType.TEXT_PLAIN)
  public String hello() {
    System.out.println(genre);
    return "Hello from RESTEasy Reactive";
  }

}

File: quarkus-tryout-kotlin/src/main/kotlin/com/example/ExampleResource.kt
package com.example

import jakarta.inject.Inject
import jakarta.ws.rs.GET
import jakarta.ws.rs.Path
import jakarta.ws.rs.Produces
import jakarta.ws.rs.core.MediaType

@Path("/hello")
class ExampleResource {

    @Inject
    lateinit var greetingConfig: GreetingConfig

    @GET
    @Produces(MediaType.TEXT_PLAIN)
    fun hello() = "${greetingConfig.message()} from RESTEasy Reactive"
}